src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java

```diff
@@ -0,0 +1,274 @@
+        requestSpec.header("Content-Type", "application/json;charset=UTF-8");
+        return requestSpec;
+    }
+
+    protected String setBodyLoanProduct() throws JsonProcessingException {
+        // Generating product name and shortname
+        String name = new StringBuilder().append("loan").append(getAlphaNumericString(4)).toString();
+        String shortName = getAlphaNumericString(4);
+
+        AllowAttributeOverrides allowAttributeOverrides = new AllowAttributeOverrides(true, true, "interest-principal-penalties-fees-order-strategy", true, true, true, true, true);
+
+        LoanProduct loanProduct = new LoanProduct("USD", "false", false, "2", "2", 2, 3, 1, 0, 6, new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), false, "1", 1, 1, true, 0, 2, 1, false, false, name, shortName, "12", "100", "3000", "21000", "3", "36", "60", "5.9", 19, "35.9", "1", true, true, "7", "90", 1, new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), "interest-principal-penalties-fees-order-strategy", allowAttributeOverrides, "en", "dd MMMM yyyy", new ArrayList<>(), false, null);
+
+        return objectMapper.writeValueAsString(loanProduct);
+    }
+
+
+    protected String setBodyLoanAccount() throws JsonProcessingException {
+
+        String date = getCurrentDate();
+        setcurrentDate(date);
+
+        CreatePayerClientResponse createPayerClientResponse = objectMapper.readValue(
+                responsePayerClient, CreatePayerClientResponse.class);
+        LoanProductResponse loanProductResponse = objectMapper.readValue(
+                responseLoanProduct, LoanProductResponse.class);
+        String clientId = createPayerClientResponse.getClientId();
+        int resourceId = Integer.parseInt(loanProductResponse.getResourceId());
+
+        LoanAccountData loanAccountData = new LoanAccountData(clientId, resourceId, new ArrayList<>(), 7800, 12, 2, 12, 1, 2, 8.9, 1, false, 0, 0, false, 7, "interest-principal-penalties-fees-order-strategy", new ArrayList<>(), currentDate, "en", "dd MMMM yyyy", "individual", currentDate, currentDate);
+
+        return objectMapper.writeValueAsString(loanAccountData);
+    }
+
+    private String getCurrentDate() {
+        Date date = new Date();
+        return new SimpleDateFormat("dd MMMM yyyy").format(date);
+    }
+
+    protected String setBodyLoanApprove(int amount) throws JsonProcessingException {
+        LoanApprove loanApprove = new LoanApprove(currentDate, amount, currentDate, new ArrayList<>(), "en", "dd MMMM yyyy");
+        return objectMapper.writeValueAsString(loanApprove);
+    }
+
+    protected Object setBodyLoanRepayment(String amount) throws JsonProcessingException {
+        LoanRepaymentDTO loanRepaymentDTO = new LoanRepaymentDTO(currentDate, "1", amount, "en", "dd MMMM yyyy");
+        return objectMapper.writeValueAsString(loanRepaymentDTO);
+    }
+
+    protected RequestSpecification setHeadersSavings(RequestSpecification requestSpec) {
```

Comment:
Delete this method and re use the one given above

Reply:
both have different cases in headers